sdk/helper/keysutil/policy.go

```diff
@@ -1335,6 +1335,8 @@ func (p *Policy) VerifySignature(context, input []byte, hashAlgorithm HashType,
 	}
 }
 
+// Rotate rotates the policy and persists it to storage.
+// If the rotation partially fails, the policy state will be restored.
```

Comment:
This comment is not correct right (and behavior remains unchanged), since we're passing `false` to `restoreOnError`?

Reply:
I'm fairly confident that the comment is correct. The logic for restoring the policy state exists in both `Rotate` and `RotateInMemory`.

In `Rotate`, we need to restore the policy state for a failure to both `RotateInMemory` and `Persist`. That's why [this snippet](https://github.com/hashicorp/vault/blob/policy-rotate-no-persist/sdk/helper/keysutil/policy.go#L1338-L1356) of code exists in both methods.

Reply:
Oh I didn't realize that Rotate had it's own restore defer logic (that was collapsed on GH).

Reply:
If we're restoring within `RotateInMemory` do we need to do that again on `Rotate` itself?

Reply:
Ya, we need to do restore in `Rotate`, since it's possible to fail at the call to `Persist`.

Reply:
This test case failed without the restore logic in `Rotate` ([policy_test.go#L461-L473](https://github.com/hashicorp/vault/blob/master/sdk/helper/keysutil/policy_test.go#L461-L473)). If the call to persist fails, it's expected that the policy state is restored.

Reply:
Ah that's right, we have to roll back on persist failure as well.